### ISSUE TITLE
Update to oddsportal website means that scraper now fails

### ISSUE THREAD

**Issue by CanProjects** [Seb943/scrapeOP#8]

**Comment by pferreira8** [comment]
Sadly dealing with this issue too. I've used this code for going on 2 years now in my nba model pipeline. I'm not the developer of this project but I will try and identify the root issue.

**Comment by ApuestasPPH** [comment]
I miss the old Odds Portal website does this code still work with the New Site?

**Comment by andrewting19** [comment]
https://codereview.stackexchange.com/questions/283181/code-scraping-oddsportal-using-beautifulsoup

Thank me later

**Comment by cardchase** [comment]
> https://codereview.stackexchange.com/questions/283181/code-scraping-oddsportal-using-beautifulsoup

Yep, I posted the question. [This code works better](https://pastebin.com/Sd0E1Hmm) as the new site is an infinite scrolling site so the more you scroll, the more odds you get.

**Comment by andrewting19** [comment]
Thanks for the code! Do you know of any way to scrape the historical odds on the results pages?

**Comment by cardchase** [comment]
> Thanks for the code! Do you know of any way to scrape the historical odds on the results pages?

Replace the page url with the page you'd like to scrape. Usually the date works fine from 1 jan 23 onwards.

**Comment by andrewting19** [comment]
Unfortunately doesn't work for links like this one (https://www.oddsportal.com/esports/usa/league-of-legends-lcs/results/). I ended up writing my own janky code to do it lol

**Comment by CanProjects** [comment]
> Unfortunately doesn't work for links like this one (https://www.oddsportal.com/esports/usa/league-of-legends-lcs/results/). I ended up writing my own janky code to do it lol

I have my own league of legends pipeline (its actually why I asked this question but i ended up writing the scraper code myself and scraping it)  but I have been unable to obtain a superior logloss to the regular betting odds, have you been able to fit a meaningful model?

**Comment by andrewting19** [comment]
Haven't tried moneyline yet. I'm using the odds as a feature for modeling kills instead.

**Comment by sword134** [comment]
> > https://codereview.stackexchange.com/questions/283181/code-scraping-oddsportal-using-beautifulsoup
> 
> Yep, I posted the question. [This code works better](https://pastebin.com/Sd0E1Hmm) as the new site is an infinite scrolling site so the more you scroll, the more odds you get.

Im trying to run this code out of the box with no modifications but it just hangs at the for game_data in game_data_results: loop. No progress is made, adding a print statement in the for loop never triggers.

**Comment by SSaltyyy** [comment]
> Unfortunately doesn't work for links like this one (https://www.oddsportal.com/esports/usa/league-of-legends-lcs/results/). I ended up writing my own janky code to do it lol

Could you post it? Or if someone could update the original scrapeOP...

**Comment by SSaltyyy** [comment]
> > Thanks for the code! Do you know of any way to scrape the historical odds on the results pages?
> 
> Replace the page url with the page you'd like to scrape. Usually the date works fine from 1 jan 23 onwards.

Result pages are without infinite scrolling so this doesn't work.

**Comment by oconbball4** [comment]
> > https://codereview.stackexchange.com/questions/283181/code-scraping-oddsportal-using-beautifulsoup
> 
> Yep, I posted the question. [This code works better](https://pastebin.com/Sd0E1Hmm) as the new site is an infinite scrolling site so the more you scroll, the more odds you get.

How do I implement this code? Looking through it not sure where to change the settings on what sport. Trying to pull league of legends historical data.
